includes/transforms.js

```diff
@@ -86,8 +86,27 @@ class SplittingStream extends Duplex {
   }
 }
 
+class SideEffect extends PassThrough {
+  constructor(fn, options) {
+    super(options);
+    this.fn = fn;
+  }
+
+  async doSideEffect(chunk) {
+    return await this.fn(chunk);
+  }
+
+  _transform(chunk, encoding, callback) {
+    this.doSideEffect(chunk).then(() => {
+      super._transform(chunk, encoding, callback);
+    }
+    ).catch((err) => { callback(err); });
```

Comment:
is linter ok with this formatting? we have opening parens on a single line, but closing on two separate with no indent, so this looks a bit weird at first glance

Reply:
It didn't complain, but agree it was weird, reformatted in https://github.com/IBM/couchbackup/pull/654/commits/dd8eb24862edee838afc354027e4dfff382ebbd1